testing/trino-product-tests/src/main/java/io/trino/tests/kafka/TestKafkaAvroReadsSmokeTest.java

```diff
@@ -205,20 +183,51 @@ public String toString()
         }
     }
 
+    private static void createAvroTable(String schemaPath, String tableName, String topicName, Map<String, Object> record, MessageSerializer messageSerializer)
+    {
+        try {
+            String schema = Files.readString(new File(schemaPath).toPath());
+            createAvroTable(new TestingAvroSchema(schema, ImmutableList.of(), ImmutableList.of()), tableName, topicName, record, messageSerializer);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static void createAvroTable(TestingAvroSchema schema, String tableName, String topicName, Map<String, Object> record, MessageSerializer messageSerializer)
+            throws IOException
+    {
+        byte[] avroData = messageSerializer.serialize(topicName, schema, record);
+
+        KafkaTableDefinition tableDefinition = new KafkaTableDefinition(
+                KAFKA_SCHEMA + "." + tableName,
+                topicName,
+                new ListKafkaDataSource(ImmutableList.of(
+                        new KafkaMessage(
+                                contentsBuilder()
+                                        .appendBytes(avroData)
+                                        .build()))),
+                1,
+                1);
+        KafkaTableManager kafkaTableManager = (KafkaTableManager) testContext().getDependency(TableManager.class, "kafka");
+        kafkaTableManager.createImmutable(tableDefinition, tableHandle(tableName).inSchema(KAFKA_SCHEMA));
+    }
+
     @FunctionalInterface
     private interface MessageSerializer
     {
-        byte[] serialize(String topic, Schema schema, Map<String, Object> values)
+        byte[] serialize(String topic, TestingAvroSchema testingAvroSchema, Map<String, Object> values)
```

Comment:
You can accept `ParsedSchema` interface instead of `TestingAvroSchema` implementation